frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -17,9 +17,17 @@ function ReaderWikipedia:init()
 end
 
 function ReaderWikipedia:onLookupWikipedia(word, box)
-    -- detect language of the text
-    local ok, lang = pcall(Translator.detect, Translator, word)
-    if not ok then return end
+    local lang_reader = G_reader_settings:readSetting("language")
+    -- set language from book properties
+    if self.view.document:getProps().language ~= nil then
```

Comment:
One more suggestion, if `self.view.document:getProps().language` is not `nil`, then you are doing an extra call to `self.view.document:getProps()` and  `G_reader_settings:readSetting("language")` here. 




Here is a more efficient way to write this logic:

```lua
local lang = self.view.document:getProps().language
if lang == nil then
    lang = G_reader_settings:readSetting("language")
    if lang == nil then
        local ok
        ok, lang = pcall(Translator.detect, Translator, word)
        if not ok then return end
    end
end
```